package/gargoyle/files/www/js/common.js

```diff
@@ -1035,7 +1071,32 @@ function setElementEnabled(element, enabled, defaultValue)
 	}
 }
 
+function setElementReadOnly(element, readOnly, associate = null)
```

Comment:
This has been previously handled just using the disabled function. What are the benefits? If this is better, should it be more generic to allow the bootstrap styles and or themes to control the look?

Reply:
This is used when the HTTP -> HTTPS redirect is selected to signify that the remote HTTPS is enabled but cannot be freely chosen. Instead, the remote HTTPS port must equal the local one since uhttpd is only aware of local ports and will redirect to the local HTTPS port even if the request came from a remote HTTP port.

Yes, it should be more generic. Instead of using dashes, I would prefer Bootstrap's [Readonly plain text](https://getbootstrap.com/docs/4.0/components/forms/#readonly-plain-text) input style as default and themes should be able to overwrite it. But it requires Bootstrap version 4.

Reply:
Moved the readonly style from JS to CSS. Now it can be overwritten by themes. Dashing is maybe not so bad, it seems "half" an input, still editable by its associated field.